### ISSUE TITLE
Duckdb read parquet error

### ISSUE THREAD

**Issue by noklam** [eakmanrq/sqlframe#201]
I cannot figure out what is happening. It seems like the error coming from `sqlglot`.

<img width="969" alt="image" src="https://github.com/user-attachments/assets/3e0753a3-ba76-4acf-b544-3c515e9047b9">
Alternatively, duckdb native API will work without problem:
```
SELECT * from read_parquet('tmp.parquet');
```

stacktrace:
```
ParseError                                Traceback (most recent call last)
Cell In[45], [line 6](vscode-notebook-cell:?execution_count=45&line=6)
      [2](vscode-notebook-cell:?execution_count=45&line=2) from sqlframe.duckdb import functions as F
      [5](vscode-notebook-cell:?execution_count=45&line=5) session = DuckDBSession()
----> [6](vscode-notebook-cell:?execution_count=45&line=6) session.read.parquet("tmp.parquet")

File /workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:265, in _BaseDataFrameReader.parquet(self, *paths, **options)
    [224](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:224) def parquet(self, *paths: str, **options: OptionalPrimitiveType) -> DF:
    [225](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:225)     """
    [226](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:226)     Loads Parquet files, returning the result as a :class:`DataFrame`.
    [227](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:227) 
   (...)
    [263](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:263)     +---+------------+
    [264](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:264)     """
--> [265](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:265)     dfs = [self.load(path=path, format="parquet", **options) for path in paths]  # type: ignore
    [266](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:266)     return reduce(lambda a, b: a.union(b), dfs)

File /workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:265, in <listcomp>(.0)
    [224](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:224) def parquet(self, *paths: str, **options: OptionalPrimitiveType) -> DF:
    [225](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:225)     """
    [226](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:226)     Loads Parquet files, returning the result as a :class:`DataFrame`.
    [227](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:227) 
   (...)
    [263](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlframe/base/readerwriter.py:263)     +---+------------+
...

File /workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlglot/__init__.py:140, in parse_one(sql, read, dialect, into, **opts)
      [0](https://vscode-remote+dev-002dcontainer-002b7b22686f737450617468223a222f776f726b7370616365732f7079636f6e686b2d32303234222c226c6f63616c446f636b6572223a66616c73652c22636f6e66696746696c65223a7b22246d6964223a312c2270617468223a222f776f726b7370616365732f7079636f6e686b2d323032342f2e646576636f6e7461696e65722f646576636f6e7461696e65722e6a736f6e222c22736368656d65223a227673636f64652d66696c65486f7374227d7d-0040ssh-002dremote-002b0192fb64-002d695f-002d7bf1-002d8f08-002dba565395d64b-002egitpod-002eremote.vscode-resource.vscode-cdn.net/workspaces/pyconhk-2024/.venv/lib/python3.10/site-packages/sqlglot/__init__.py:0) <Error retrieving source code with stack_data see ipython/ipython#13598>

ParseError: No expression was parsed from 'TIMESTAMP_NS'

```

**Comment by eakmanrq** [comment]
@noklam Sorry for the delay in response. Can you share a parquet file that reproduces this issue? The parquet file I have for testing loads fine with DuckDB so I want to get a file that reproduces the issue first. 

An alternative would be providing steps to produce a parquet file that reproduces the issue.

**Comment by noklam** [comment]
@eakmanrq Thanks for getting back. Please change the suffix to `parquet` as Github wouldn't allow me to upload a parquet file. I created a minimal file with 5 rows.
[small_parquet.json](https://github.com/user-attachments/files/17810922/small_parquet.json)

**Comment by eakmanrq** [comment]
Thanks @noklam I was able to easily repro with what you provided. Fix will be release in `3.8.1`.
